075-connector-offsets-support.md

```diff
@@ -0,0 +1,270 @@
+* GET /connectors/{connector}/offsets
+* PATCH /connectors/{connector}/offsets
+* DELETE /connectors/{connector}/offsets
+* PUT /connectors/{connector}/stop
+
+Support for the "stop" endpoint was added via [PR 9095](https://github.com/strimzi/strimzi-kafka-operator/pull/9095).
+Currently users that are utilising the KafkaConnector resource cannot make use of the API endpoints for managing offsets.
+
+## Motivation
+
+Previously if users wanted to list, update or delete offsets for a connector they had to interact with the relevant consumer groups for sink connectors or directly write to the Kafka Connect internal offsets topic for source connectors.
+[KIP-875][kip] adds new endpoints to let the user do this more easily.
+There are multiple reasons a user might want to manage offsets, for example (as listed in the KIP):
+* Resetting the offsets for a connector while iterating rapidly in a development environment (so that the connector does not have to be renamed each time)
+* Viewing the in-memory offsets for source connectors on the cluster in order to recover from accidental deletion of the Connect source offsets topic (this is currently possible with the config topic, and can be a life-saver)
+* Monitoring the progress of running connectors on a per-source-partition basis 
+* Skipping records that cause issues with a connector and cannot be addressed using existing error-handling features
+
+## Proposal
+
+Strimzi should include mechanisms to let users manage offsets when utilising the KafkaConnector resources.
+This proposal suggests introducing a new annotation on the KafkaConnector resource to trigger each operation.
+The operations are:
+* List offsets.
+* Alter offsets.
+* Reset offsets.
+
+The list offsets operation uses a ConfigMap for the output.
+The alter offsets operation uses a ConfigMap for the input.
+
+### Request/Response from the API endpoints
+
+The response body received when using the GET endpoint, and the request body used when calling the PATCH endpoint have identical structures to allow users to easily fetch and then alter offsets.
+The structure of the body is different for source and sink connectors.
+This is because sink connectors use regular Kafka offsets, whereas source connectors determine their own structure for offsets.
+This difference is reflected in the approaches outlined in this proposal.
+
+Source connector format is:
+```
+{
+  "offsets": [
+    {
+      "partition": {
+        // Connector-defined source partition
+      },
+      "offset": {
+        // Connector-defined source offset
+      }
+    }
+  ]
+}
+```
+
+Sink connector format is:
+```
+{
+  "offsets": [
+    {
+      "partition": {
+        "kafka_topic": // Kafka topic
+        "kafka_partition": // Kafka partition
+      },
+      "offset": {
+        "kafka_offset": // Kafka offset
+      }
+    }
+  ]
+}
+```
+
+### New annotation
+
+All three actions will make use of a new annotation called `strimzi.io/connectorOffsets`.
```

Comment:
most of our annotations are using a format like this `strimzi.io/connector-offsets`. I would change this way (here and of course everywhere it's mentioned again).

Reply:
+1 camel case looks very weird here

Reply:
Good point, I've updated it